[Test] Creating CADMIN 1.19 python3 test module

j-ororke opened the pull request:
Creating python3 TC CADMIN 1.19 test module:
- Test module created following steps in PR: [4669](https://github.com/CHIP-Specifications/chip-test-plans/pull/4669)
- Removed yaml test script test_TC_CADMIN_1_19.yaml as no longer needed

j-ororke committed:
[Test] Creating CADMIN 1.19 test module:
- Test module created following steps in PR: https://github.com/CHIP-Specifications/chip-test-plans/pull/4669

restyled-commits committed:
Restyled by autopep8

j-ororke committed:
Update TC_CADMIN_1_19 test module:
- Resolving linting issues

j-ororke committed:
Removing yaml test module for TC_CADMIN_1_19:
- Removed yaml test as we have created python test module in its place

j-ororke committed:
Update TC_CADMIN_1_19.py

- Removed debugging print statements as no longer needed in test code

cecille commented on src/python_testing/TC_CADMIN_1_19.py:
```diff
@@ -0,0 +1,201 @@
+    async def CommissionAttempt(
+            self, setupPinCode: int, thnum: int, th: str, fail: bool):
+
+        logging.info(f"-----------------Commissioning with TH_CR{str(thnum)}-------------------------")
```
```suggestion
        logging.info(f"-----------------Commissioning with TH_CR{thnum}-------------------------")
```

cecille commented on src/python_testing/TC_CADMIN_1_19.py:
```diff
@@ -0,0 +1,201 @@
+
+
+class TC_CADMIN_1_19(MatterBaseTest):
+    async def OpenCommissioningWindow(self) -> CommissioningParameters:
```
can you use the openCommissionignWindow function in matter_testing instead of re-implementing?

j-ororke replied:
I have updated the code to remove the OpenCommissioningWindow and CommissionAttempt local class functions to using the MatterBaseTest version of those here now. 
Also, added the timeout variable to allow dynamic setting of now instead of statically set.

cecille commented on src/python_testing/TC_CADMIN_1_19.py:
```diff
@@ -0,0 +1,201 @@
+            setupPinCode = params.setupPinCode
+
+            self.step("5b")
+            fids_ca_dir['thca' + str(fid)] = self.certificate_authority_manager.NewCertificateAuthority()
```
why are you using strings as keys rather than just numbers if they're separate dicts?

j-ororke replied:
So, I had been using it to make the names a bit easier possibly to read the names, but since not needed I have changed those to using just the range number now for the dictionary key names.

cecille commented on src/python_testing/TC_CADMIN_1_19.py:
```diff
@@ -0,0 +1,201 @@
+        # TH reads the CommissionedFabrics attributes from the Node Operational Credentials cluster
+        current_fabrics = await self.read_single_attribute_check_success(dev_ctrl=self.th1, fabric_filtered=False, endpoint=0, cluster=OC_cluster, attribute=OC_cluster.Attributes.SupportedFabrics)
+        if current_fabrics != max_fabrics:
+            asserts.fail(f"Expected number of fabrics not correct, instead was {str(current_fabrics)}")
```
```suggestion
            asserts.fail(f"Expected number of fabrics not correct, instead was {current_fabrics}")
```

cecille commented on src/python_testing/TC_CADMIN_1_19.py:
```diff
@@ -0,0 +1,201 @@
+            self.step("5b")
+            fids_ca_dir['thca' + str(fid)] = self.certificate_authority_manager.NewCertificateAuthority()
+            fids_fa_dir['thfa' + str(fid)] = fids_ca_dir['thca' + str(fid)].NewFabricAdmin(vendorId=0xFFF1, fabricId=fid)
+            fids['th' + str(fid)] = fids_fa_dir['thfa' + str(fid)].NewController(nodeId=fid, useTestCommissioner=True)
```
You don't need to use the test commissioner here.

j-ororke replied:
Thank you, this has been removed from the test script

j-ororke committed:
Update src/python_testing/TC_CADMIN_1_19.py

Co-authored-by: C Freeman <cecille@google.com>

j-ororke committed:
Update src/python_testing/TC_CADMIN_1_19.py

Co-authored-by: C Freeman <cecille@google.com>

j-ororke committed:
Merge branch 'master' into create_CADMIN_1_19_test

j-ororke committed:
Updating TC_CADMIN_1_19 and matter_testing scripts:
- Updated CI Arguments
- Updated dependency location for matter_testing_support
- Removed local OpenCommissioningWindow function, updated to using MatterBaseTest version
- Removed local CommissioningAttempt function, moved to using function call directly in test
- Removed initialization of TH2 controller as not needed
- Removed additional dictionary names and delimited it down to just using numbers instead.
- Updated matter_testing support script to include setting timeout value dynamically in openCommissioningWindow function

restyled-commits committed:
Restyled by autopep8

j-ororke committed:
Update TC_CADMIN_1_19.py

Resolving Linting errors

cecille commented on src/python_testing/TC_CADMIN_1_19.py:
```diff
@@ -0,0 +1,175 @@
+                filterType=ChipDeviceCtrl.DiscoveryFilterType.LONG_DISCRIMINATOR, filter=params.randomDiscriminator)
+
+        except ChipStackError as e:
+            self.print_step("err", e.err)
```
remove print - think this might be leftover debug stuff.

j-ororke replied:
thank you, I have removed this debugging print statement.

j-ororke committed:
Update TC_CADMIN_1_19.py

- Moved location of print_step to top of for range for creating virtual fragments
- Removed unnecessary print_step for catching error

j-ororke commented on src/python_testing/TC_CADMIN_1_19.py:
```diff
@@ -0,0 +1,175 @@
+                nodeId=self.dut_node_id, setupPinCode=params.commissioningParameters.setupPinCode,
+                filterType=ChipDeviceCtrl.DiscoveryFilterType.LONG_DISCRIMINATOR, filter=params.randomDiscriminator)
+
+            self.print_step("commissioning iteration", fid)
```
Thank you, I have updated the location of that print statement to the top of the for loop

tehampson commented on src/python_testing/TC_CADMIN_1_19.py:
```diff
@@ -0,0 +1,173 @@
+            # Make sure that current test step is 5, resets here after each loop
+            self.current_step_index = 5
+
+            self.step("5a")
```
Can you now call this multiple times with the same value? Previously I recall that you could only call step on the next value and couldn't actually repeat

j-ororke replied:
Hi @tehampson ,
So, you sort of can, it seems from my testing you just have to make the script repeat the prior step first, so when you utilize self.current_step_index and reset the value to the previous step it appears to work.

tehampson commented on src/python_testing/TC_CADMIN_1_19.py:
```diff
@@ -0,0 +1,173 @@
+        fids_fa_dir = {}
+        fids = {}
+        self.print_step("total fabrics", max_fabrics - initial_number_of_fabrics)
+        for fid in range(1, max_fabrics - initial_number_of_fabrics):
```
Shouldn't range start at 0? If `max_fabrics - initial_number_of_fabrics` is 1 you will currently loop through this 0 times which seems incorrect

j-ororke replied:
You are correct, when I made a prior change this appears to have gotten changed by accident.
I have updated this to start at int 0 rather than start at int 1.

tehampson commented on src/python_testing/TC_CADMIN_1_19.py:
```diff
@@ -0,0 +1,173 @@
+        max_fabrics = await self.read_single_attribute_check_success(dev_ctrl=self.th1, fabric_filtered=False, endpoint=0, cluster=OC_cluster, attribute=OC_cluster.Attributes.SupportedFabrics)
+
+        self.step(5)
+        fids_ca_dir = {}
+        fids_fa_dir = {}
+        fids = {}
```
Are these supposed to be dictionaries or arrays (list). You are using an index later on so it seems like they should an array starting at 0

tehampson commented on src/python_testing/TC_CADMIN_1_19.py:
```diff
@@ -0,0 +1,173 @@
+
+        self.step(8)
+        # TH creates a controller on a new fabric and commissions DUT_CE using that controller
+        fids_ca_dir[current_fabrics + 1] = self.certificate_authority_manager.NewCertificateAuthority()
```
I would set a variable to `current_fabrics + 1`, from there I would use that when accessing the list

j-ororke replied:
That is a great point, I have updated to setting a variable as you mentioned in your comment above, then using it throughout the rest of the test for current_fabrics + 1

tehampson commented on src/python_testing/TC_CADMIN_1_19.py:
```diff
@@ -0,0 +1,173 @@
+        self.step(10)
+        # TH reads the CommissionedFabrics attributes from the Node Operational Credentials cluster.
+        current_fabrics = await self.read_single_attribute_check_success(dev_ctrl=self.th1, fabric_filtered=False, endpoint=0, cluster=OC_cluster, attribute=OC_cluster.Attributes.CommissionedFabrics)
+        if current_fabrics != initial_number_of_fabrics:
```
Why not just use asserts.assert_equals(current_fabrics, initial_number_of_fabrics)

tehampson replied:
Same comment for all the current if statements that then have an asserts.fail after that

j-ororke replied:
This is a good idea! 
I have changed to using this asserts method in place of using the if statements.

j-ororke committed:
Updated TC_CADMIN_1_19 test module:
	- Created new variable for current_fabric + 1
	- Replaced if statements with asserts.assert_equal()
	- Changed start of range from 1 to 0
	- Changed test steps to match test plan better, removed test step 1

restyled-commits committed:
Restyled by autopep8

j-ororke commented on src/python_testing/TC_CADMIN_1_19.py:
```diff
@@ -0,0 +1,173 @@
+        self.th1 = self.default_controller
+
+        self.step(2)
+        GC_cluster = Clusters.GeneralCommissioning
```
Thank you, 
I have updated the test steps in the test module here now to match with the test plan.

tehampson replied:
Lets leave this comment open until the testplan has been updated just to make sure the two are aligned before the PR merges

cecille replied:
This is now merged.

j-ororke committed:
Updated TC_CADMIN_1_19 test module:
- Added new criteria to test step 3 where max_fabrics is checked to make sure it is greater than initial_fabrics count

j-ororke committed:
Updated TC_CADMIN_1_19 test module:
- Removed print statement from debugging session

restyled-commits committed:
Restyled by autopep8
